api/model/api.go

```diff
@@ -69,7 +69,7 @@ const (
 )
 
 // APIProtocol is the network transport protocol supported by the gateway
-// +kubebuilder:validation:Enum=h2c;tcp;tls;http;https;
+// +kubebuilder:validation:Enum="";h2c;tcp;tls;http;https;
```

Comment:
Could you explain why we are permitting empty value here for APIProtocol?

Reply:
Dashboard stores empty value for `http` APIProtocol. Once `http` APIs are exported from Dashboard, the corresponding YAML file has an empty APIProtocol value (`""`), which prevents applying that manifest directly due to validation failure. Users need to update empty values with `http`. That's why I added empty string, `""`, to the validation enum.

Reply:
Does it make sense to mutate the response from the Dashboard API to insert http in there? Perhaps we could fix it inside the defaulting webhook? 

Or with this change are we saying that this field is completely optional - and up to the dashboard / gateway API to decide what to do with it?

Reply:
>Or with this change are we saying that this field is completely optional - and up to the dashboard / gateway API to decide what to do with it?

Actually, this is what I was trying to achieve - letting dashboard/gw decide on what to do with empty "protocol" since providing empty protocol does not raise any errors on the dashboard/gw side.

As you mentioned, we can update empty APIProtocol values in defaulting webhooks but I believe it might cause a problem since the dashboard has a mechanism to set it to a default value. It defaults `protocol` to your gw configuration if `protocol` is empty.

If you see this change as breaking, I can update snapshot tool logic to update empty APIProtocols as `http`.

Reply:
I think it's a good decision. Thanks.